packages/drivers/debugger/src/fluidDebuggerController.ts

```diff
@@ -310,3 +337,39 @@ export class DebugReplayController extends ReplayController implements IDebugger
         this.startSeqDeferred.resolve(seq);
     }
 }
+
+async function* generateSequencedMessagesFromDeltaStorage(deltaStorage: IDocumentDeltaStorageService)  {
+    const timeStart = Date.now();
+    let requests = 0;
+    let opsStorage = 0;
+    let lastSeq = 0;
+    const batch = 2000;
+    while (true) {
+        requests++;
+        const messages = await loadChunk(lastSeq, lastSeq + batch, deltaStorage);
+        if (messages.length === 0) {
+            break;
+        }
+        yield messages;
```

Comment:
Given this is single use utility, you can simply accumulate all ops and return them, not return chunks.
Chunks are good for faster processing when chunks can be processed in parallel with downloading next chunk

Reply:
Make sense.  But related, this chunk was copy/pasted from the fetch tool.  Ideally we'd share it.  Suggestions on where? (Or to deliberately have another implementation?) 

I would like to incorporate stats here at some point, so will definitely want to share that.

Reply:
Ideally this is shared with logic in DetlaManager as well (that's where I could not use async generators in the past, so logic is slightly more twisted there). I think the right way to do it is to open an issue for all of that. This logic likely needs to be moved to driver level.
+ @ChumpChief as FYI

Reply:
Makes sense, logged an issue here: https://github.com/microsoft/FluidFramework/issues/3833.